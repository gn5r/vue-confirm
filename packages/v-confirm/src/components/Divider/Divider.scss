@use "../../styles/theme" as *;

.v-confirm__divider {
  display: block;
  flex: 1 1 0px;
  max-width: 100%;
  height: 0;
  max-height: 0;
  border: solid;
  border-width: thin 0 0 0;
  transition: inherit;
  overflow: visible;

  @at-root {
    @include theme(divider) using ($theme) {
      border-color: map-get($theme, "divider");
    }
  }
}
