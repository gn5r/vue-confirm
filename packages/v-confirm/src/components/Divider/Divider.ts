// Mixins
import Themeable from "../../mixins/themeable";

// Types
import { VNode } from "vue/types";

// Style
import "./Divider.scss";

export default Themeable.extend({
  name: "divider",
  computed: {
    classes(): object {
      return {
        divider: true,
        ...this.themeClasses,
      };
    },
  },
  render(h): VNode {
    return h("hr", {
      attrs: { role: "separator", "area-orientation": "horizontal" },
      class: this.classes,
    });
  },
});
