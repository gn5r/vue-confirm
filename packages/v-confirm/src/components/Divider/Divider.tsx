import { defineComponent, computed } from "vue";
import { useThemeProps, useThemeClasses } from "@/composables/theme";
import "./Divider.scss";

export default defineComponent({
  name: "divider",
  props: useThemeProps(),
  setup(props) {
    const themeClasses = useThemeClasses(props);
    return () => (
      <>
        <hr
          role="separator"
          aria-orientation="horizontal"
          class={["divider", themeClasses.value]}
        />
      </>
    );
  },
});
