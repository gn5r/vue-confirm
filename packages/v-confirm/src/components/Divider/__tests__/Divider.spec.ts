import { createLocalVue, mount, MountOptions } from "@vue/test-utils";
import Divider from "../Divider";

describe("Divider.ts", () => {
  const localVue = createLocalVue();

  type Instance = InstanceType<typeof Divider>;
  const mountFunc = (options?: MountOptions<Instance>) =>
    mount(Divider, { localVue, ...options });

  it("should render", () => {
    const wrapper = mountFunc();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("render a dark component", () => {
    const wrapper = mountFunc({
      propsData: { dark: true },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
