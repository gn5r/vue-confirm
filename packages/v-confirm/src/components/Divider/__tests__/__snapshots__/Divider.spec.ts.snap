// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider.ts render a dark component 1`] = `<hr role="separator" area-orientation="horizontal" class="dark divider">`;

exports[`Divider.ts should render 1`] = `<hr role="separator" area-orientation="horizontal" class="light divider">`;
